src/components/ContentHighlights/data/hooks.js

```diff
@@ -30,12 +33,41 @@ export function useHighlightSetsForCuration(enterpriseCuration) {
   return highlightSets;
 }
 
+export function useHighlightSet(highlightSetUUID) {
+  const [highlightSet, setHighlightSetItems] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const getHighlightSetItems = useCallback(async () => {
+    try {
+      const { data } = await EnterpriseCatalogApiService.fetchHighlightSet(highlightSetUUID);
+      const result = camelCaseObject(data);
+      setHighlightSetItems(result);
+    } catch (e) {
+      setError(e);
+      logError(e);
+    } finally {
+      setIsLoading(false);
+    }
+  }, [highlightSetUUID]);
+
+  useEffect(() => {
+    getHighlightSetItems();
+  }, [getHighlightSetItems]);
+
+  return {
+    highlightSet,
+    isLoading,
+    error,
+  };
+}
+
 /**
  * Defines an interface to mutate the `ContentHighlightsContext` context value.
  */
 export function useContentHighlightsContext() {
   const setState = useContextSelector(ContentHighlightsContext, v => v[1]);
-
+  const getState = useContextSelector(ContentHighlightsContext, v => v[0]);
```

Comment:
With `useContextSelector`, the 2nd argument is generally extracting the specific data from the state, not the entire state value itself. That way, it's smart enough to only subscribe to changes from a "slice" of state, rather than the entire state value, e.g.:

```jsx
const currentSelectedRowIds = useContextSelector(ContentHighlightsContext, v => v[0]. stepperModal.currentSelectedRowIds);
```